tests/functional/adapter/test_basic_iceberg.py

```diff
@@ -0,0 +1,147 @@
+"""
+Run the basic dbt test suite on hive tables when applicable.
+
+Some test classes are not included here, because they don't contain table models.
+Those are run in the hive test suite.
+"""
+import pytest
+
+from dbt.tests.adapter.basic.files import (
+    base_ephemeral_sql,
+    base_materialized_var_sql,
+    base_table_sql,
+    base_view_sql,
+    config_materialized_incremental,
+    config_materialized_table,
+    ephemeral_table_sql,
+    ephemeral_view_sql,
+    generic_test_table_yml,
+    generic_test_view_yml,
+    incremental_sql,
+    model_base,
+    model_ephemeral,
+    schema_base_yml,
+)
+from dbt.tests.adapter.basic.test_adapter_methods import BaseAdapterMethod
+from dbt.tests.adapter.basic.test_base import BaseSimpleMaterializations
+from dbt.tests.adapter.basic.test_ephemeral import BaseEphemeral
+from dbt.tests.adapter.basic.test_generic_tests import BaseGenericTests
+from dbt.tests.adapter.basic.test_incremental import BaseIncremental
+from dbt.tests.adapter.basic.test_snapshot_check_cols import BaseSnapshotCheckCols
+from dbt.tests.adapter.basic.test_snapshot_timestamp import BaseSnapshotTimestamp
+
+iceberg_config_materialized_table = """
+  {{ config(materialized="table", table_type="iceberg") }}
+"""
+
+iceberg_config_materialized_incremental = """
+  {{ config(materialized="incremental", table_type="iceberg", incremental_strategy="merge", unique_key="id") }}
+"""
+
+iceberg_model_base = """
+  select
+      id,
+      name,
+      {{ cast_timestamp('some_date') }} as some_date
+  from {{ source('raw', 'seed') }}
+"""
+
+iceberg_model_ephemeral = """
+  select
+      id,
+      name,
+      {{ cast_timestamp('some_date') }} as some_date
+  from {{ ref('ephemeral') }}
+"""
+
+
+def replace_single_model(model):
```

Comment:
nit: let's consider to rename this to something more meanifull e.g. `prepare_iceberg_models_config` or `get_iceberg_model_configs`

Reply:
Good point. Fixed it 👍